app/src/main/java/org/eyeseetea/malariacare/domain/usecase/pull/DownloadMediaUseCase.java

```diff
@@ -68,6 +70,11 @@ public void onError(FileDownloadException ex) {
                         }
 
                         @Override
+                        public void onProgress() {
+                            mCallback.onProgress();
```

Comment:
callback invocations should be executed in main thread using MainExecutor as other use cases